### ISSUE TITLE
New Pods are not created because of stuck informers

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#8528]
It looks like in some situations, the informers get stuck and as a result, the `StrimziPodSetController` does not operate the Pods anymore. This is a problem because it does not recreate them after they are deleted etc. It is not clear what the cause is and whether it is related to Strimzi, Fabric8 or to the user environment. This issue should try to track the issues and try to find common cases.

I also asked in the Fabric8 discussions for any advice to debug this: https://github.com/fabric8io/kubernetes-client/discussions/5152

**Comment by scholzj** [comment]
One of the environment where this seems to happen is `AKS on k8s 1.23.8` - reported in https://cloud-native.slack.com/archives/CMH3Q3SNP/p1684434853354859

**Comment by cjyar** [comment]
Also observed on Google Kubernetes Engine.
```
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.8-gke.500", GitCommit:"f117e29cb87cfb7e1de32ab4e163fb01ac5d0af9", GitTreeState:"clean", BuildDate:"2023-03-23T10:22:38Z", GoVersion:"go1.19.7 X:boringcrypto", Compiler:"gc", Platform:"linux/amd64"}
```

Observed behavior: After deleting a zookeeper pod, the operator isn't re-creating it. Operator logs:
```
2023-05-23 17:28:35 ERROR Util:166 - Reconciliation #9548(timer) Kafka(kafka/test1): Exceeded timeout of 300000ms while waiting for Pods resource test1-zookeeper-0 in namespace kafka to be ready
2023-05-23 17:28:35 ERROR AbstractOperator:260 - Reconciliation #9548(timer) Kafka(kafka/test1): createOrUpdate failed
io.strimzi.operator.common.operator.resource.TimeoutException: Exceeded timeout of 300000ms while waiting for Pods resource test1-zookeeper-0 in namespace kafka to be ready
...followed by a stack trace
```

Operator version: quay.io/strimzi/operator:0.35.0

**Comment by scholzj** [comment]
We prepared 0.35.1-rc1 with updated Kubernetes Client which should hopefully help with this. If you are affected by this issue, please give it a try and let us know if it helped: https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.35.1-rc1 ... I will keep this issue open until hearing more.

**Comment by cjyar** [comment]
@scholzj Thanks! So far, I'm not able to reproduce the bug with `0.35.0`, so I'm not able to comment if `0.35.1-rc1` fixes it.

**Comment by scholzj** [comment]
Ok, 0.35.1 is out ... so please upgrade and let's see if it helps ... hopefully it will.

**Comment by Aaron-ML** [comment]
@scholzj  Nice work, we will take a look at 0.35.1 and see if we can reproduce the old behavior.

**Comment by maartengo** [comment]
It seems like 0.35.1 has fixed the issue. I haven't been able to reproduce the behavior on any environment. It used to be quite reproducible after a day of uptime, but it continues to work even after 4-6 days.

**Comment by scholzj** [comment]
@maartengo That is good news. Thanks.

**Comment by scholzj** [comment]
BAsed on the reports, this seems to be fixed from 0.35.1 on.
